feat(mui): configurable `anchorOrigin` for `useNotificationProvider`

Yash-271120 opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://refine.dev/docs/guides-concepts/contributing/#commit-convention

## Bugs / Features

- [x] Related issue(s) linked
- [x] Tests for the changes have been added
- [ ] Docs have been added / updated
- [ ] Changesets have been added https://refine.dev/docs/guides-concepts/contributing/#creating-a-changeset

## What is the current behavior?
Anchor position,preventDuplicate,disableWindowBlurListener are hard coded in mui useNotificationsProvider

## What is the new behavior?
these values can now be provided as per usage

fixes #5847 

## Notes for reviewers

<!-- Add any notes/questions you may have for reviewers -->

aliemir reviewed (changes requested):
Hey @Yash-271120, thank you for the PR. I think there are some misunderstandings about this issue. Let me try to clarify the changes needed.

- You've made a change to the `notificationProvider` type in `@refinedev/core`. We don't need to have anything changed here.
- `useNotificationProvider` of `@refinedev/mui` should not accept any more parameters. Only the props like `anchorOrigin`, `disableWindowBlurListener` etc. will be replaced by the context of `RefineSnackbarProvider`.
- The issue requires changes to the `<RefineSnackbarProvider />` component. We need to pass these default values as default props to the `<RefineSnackbarProvider />`. Then those will replace the hardcoded values of `useNotificationProvider`.

Let me know if you can update this PR to reflect the changes properly 🙏

Yash-271120 commented:
@aliemir what if we remove the harcoded code for `anchorOrigin`, `disableWindowBlurListener` and let the default values of these props be handled by the snackbar from notistack itself?

aliemir commented:
> @aliemir what if we remove the harcoded code for `anchorOrigin`, `disableWindowBlurListener` and let the default values of these props be handled by the snackbar from notistack itself?

That was actually what I was trying to say 😅 These values will be set as default props of `<RefineSnackbarProvider />` and users can override them by passing explicit props.

Can you make the required changes in this PR? I'll mark this PR to be included in our May release if you can provide an update 🙏

Yash-271120 commented:
Hey @aliemir 
Sorry I misunderstood previously. Getting back at react after long time. I have updated the PR:- reverted the previous changes and removed the hard coded values and also have updated the tests
